### PR TITLE
graphql: use field resolvers

### PR DESCRIPTION
**Pull request by stefanhengl**
By using field resolvers (https://github.com/graph-gophers/graphql-go#resolvers) we can avoid a lot of boilerplate. In this PR I set the option `UseFieldResolvers` and rewrite a stub resolver that I added recently.

## Test plan
- GraphQL compiles without errors.

### CODE REVIEW

File: cmd/frontend/graphqlbackend/graphqlbackend.go [hunk 1]
```diff
@@ -626,6 +626,7 @@ func NewSchema(
 			logger: logger,
 		}),
 		graphql.UseStringDescriptions(),
+		graphql.UseFieldResolvers(),
```
Comment (mrnugget) [inline cmd/frontend/graphqlbackend/graphqlbackend.go:629]: How sure are we that this doesn't involuntarily break things?

**Comment by eseliger** [comment]
In the name of consistency - should we do this if we only use it for a single resolver? The reduction in boilerplate here is fairly minimal so far, and we'll still have to mix field and method resolvers for most other resolvers where some fields are computed, not sure if that could be confusing. WDYT?

File: cmd/frontend/graphqlbackend/graphqlbackend.go (hunk 1 above)
Comment (keegancsmith) [reply cmd/frontend/graphqlbackend/graphqlbackend.go:629]: Its so weird, I swear we used to do this. I would be interested in knowing why we stopped enabling it. I think if CI passes we can be relatively confident it works?

**Comment by keegancsmith** [comment]
I think a vast majority of resolvers would be much simpler if we had the option of field resolvers.

**Comment by mrnugget** [comment]
> I think a vast majority of resolvers would be much simpler if we had the option of field resolvers.

Who's going to rewrite them though?

**Comment by stefanhengl** [comment]
The intention was not to trigger a rewrite but to provide an option for a simpler design moving forward -> Less interfaces and less boilerplate methods for our resolvers.

However, although the schema parses successfully and Sourcegraph starts up, the tests fail.

All of the examples I looked at are related to GraphQL unions and interfaces. To me this looks like a bug in the GraphQL library we use. [I can reproduce this behaviour locally with a small test rig](https://gist.github.com/stefanhengl/52193b62d3880de4cd2b2a11a63d8eea). A fix would be to replace some of our interfaces, EG those that represent unions, with concrete types.

I will timebox this to see how extensive the change would be.

**Comment by stefanhengl** [comment]
Closing this. Turns out we would have to move a lot of code around (moving types to `graphqlbackend`) and I don't think this is worth it right now.

**Pull request closed without merging**